### DISCUSSION TITLE
[SOLVED] ERROR caused by old metadata format: ERROR render of "page" failed: execute of template failed: template: _internal/opengraph.html:36:44: balabala...

### DISCUSSION THREAD

**Discussion by gitaeron** [Lednerb/bilberry-hugo-theme#469] [General]
## First of all

Help me with the longest title after reading this, Pls (。>︿<)_θ

I'm not sure if I delete error info in the title that make others unable to searched this Error....

---

## Error with using series

If you are maintained your own bilberry-hugo-theme and had updated your theme from an older version recently, 

and you are using **Series**,

you might got this error when you run `hugo` or `hugo server` to generate your static site page:

- ERROR 2023/01/05 00:24:37 render of "page" failed: execute of template failed: template: _internal/opengraph.html:36:44: executing "_internal/opengraph.html" at <.>: range can't iterate over YOUR_SERIES_NAME

Just early than **Issue-459: Fix Open Graph metadata**  #463  at HEAD 0971422110920f48f6f4bb6207ef878eccfa4177, 
you will got this error because the new metadata format.

---

## Fixed 

It's a simple thing to fix it, it caused by the new format of metadata:

- Old =>  series: 'Your Series Name'

- New => series: [ 'Your Series Name' ]

As you can see here, just use series with `[]` character contained.

And here's something new in the new metadata, just watch [Bilberry Hugo Theme Guidence](https://themes.gohugo.io/themes/bilberry-hugo-theme/) for more info.

Here's the short brief below:

```python
title: "Open Graph Metadata with Extra Front Matter Variables"
date: 2022-12-19T19:00:00-05:00
publishDate: 2022-12-19T20:00:00-05:00
lastmod: 2022-12-19T21:00:00-05:00
images: ["/img/content/article/open-graph-metadata-with-extra-front-matter-variables/thumbnail.jpg"]
audio: "/audio/icq-remix.mp3"
videos: ["/video/test_mp4_video.mp4"]
series: ["Your Series Name"]
```

Notice: [The official Guidence](https://themes.gohugo.io/themes/bilberry-hugo-theme/#open-graph-metadata) in `open-graph-metadata` have no this description about series: ["Your Series Name"], so I add it to here in the last line. 

---

This is the error log below:

``` shell
$ hugo server

Start building sites …

hugo v0.105.0-0e3b42b4a9bdeb4d866210819fc6ddcf51582ffa+extended windows/amd64 BuildDate=2022-10-28T12:29:05Z VendorInfo=gohugoio

ERROR 2023/01/05 00:24:37 render of "page" failed: execute of template failed: template: 
_internal/opengraph.html:36:44: executing "_internal/opengraph.html" at <.>: range can't iterate over YOUR_SERIES_NAME

ERROR 2023/01/05 00:24:37 render of "page" failed: execute of template failed: template: 
_internal/opengraph.html:36:44: executing "_internal/opengraph.html" at <.>: range can't iterate over YOUR_SERIES_NAME

ERROR 2023/01/05 00:24:37 render of "page" failed: execute of template failed: template: 
_internal/opengraph.html:36:44: executing "_internal/opengraph.html" at <.>: range can't iterate over YOUR_SERIES_NAME

ERROR 2023/01/05 00:24:37 render of "page" failed: execute of template failed: template: 
_internal/opengraph.html:36:44: executing "_internal/opengraph.html" at <.>: range can't iterate overYOUR_SERIES_NAME

Error: Error building site: failed to render pages: render of "page" failed: execute of template failed: template: _internal/opengraph.html:36:44: executing "_internal/opengraph.html" at <.>: range can't iterate over YOUR_SERIES_NAME
```

**Comment by gitaeron** [comment]
If you got Error when you using bilberry hugo theme, just calm down and check everything that needed. 

And if it's out of your range, just start a Discussion and seek for help from others, and searched some key words first is a good idea.

**Comment by igor-baiborodine** [comment]
@gitaeron, the **Open Graph Metadata** section in the `README` guide was updated accordingly: 

![image](https://user-images.githubusercontent.com/1027701/210658435-daa88d1d-ca66-41e2-ac8c-fc0402baac9a.png)
